modules/msg/option_inc/default_cfe_msg_hdr_pri.h

```diff
@@ -0,0 +1,94 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/**
+ * @file
+ *
+ * Define cFS standard full header
+ *  - Avoid direct access for portability, use APIs
+ *  - Used to construct message structures
+ */
+
+#ifndef DEFAULT_CFE_MSG_HDR_PRI_H
+#define DEFAULT_CFE_MSG_HDR_PRI_H
+
+/*
+ * Include Files
+ */
+
+#include "common_types.h"
+#include "ccsds_hdr.h"
+#include "cfe_msg_sechdr.h"
+#include "cfe_msg_api_typedefs.h"
+
+/*
+ * Type Definitions
+ */
+
+/**********************************************************************
+ * Structure definitions for full header
+ *
+ * These are based on historically supported definitions and not
+ * an open source standard.
+ */
+
+/**
+ * \brief Full CCSDS header
+ */
+typedef struct
+{
+    CCSDS_PrimaryHeader_t Pri; /**< \brief CCSDS Primary Header */
+} CCSDS_SpacePacket_t;
+
+/**
+ * \brief cFS generic base message
+ *
+ * This provides the definition of CFE_MSG_Message_t
+ */
+union CFE_MSG_Message
+{
+    CCSDS_SpacePacket_t CCSDS;                             /**< \brief CCSDS Header (Pri or Pri + Ext) */
+    uint8               Byte[sizeof(CCSDS_SpacePacket_t)]; /**< \brief Byte level access */
+};
+
+/**
+ * \brief cFS command header
+ *
+ * This provides the definition of CFE_MSG_CommandHeader_t
+ */
+struct CFE_MSG_CommandHeader
+{
+    CFE_MSG_Message_t                Msg; /**< \brief Base message */
+    CFE_MSG_CommandSecondaryHeader_t Sec; /**< \brief Secondary header */
+};
+
+/**
+ * \brief cFS telemetry header
+ *
+ * This provides the definition of CFE_MSG_TelemetryHeader_t
+ */
+struct CFE_MSG_TelemetryHeader
+{
+    CFE_MSG_Message_t                  Msg;      /**< \brief Base message */
+    CFE_MSG_TelemetrySecondaryHeader_t Sec;      /**< \brief Secondary header */
+    uint8                              Spare[4]; /**< \brief Padding to end on 64 bit boundary */
```

Comment:
CFE_MSG_TelemetryHeader inserts a spare 4 bytes at the end in order to "end on a 64-bit boundary" - will this necessarily be the case all the time? In the classic API this structure used to be aligned on a 32-bit boundary. The header in total was 12 bytes and we add another 4 for a total of 16 bytes. If the structure is aligned to a 32-bit boundary as in the classic API, this could either end on a 32-bit boundary or a 64-bit boundary. If it isn't aligned, then it seems that it could end at a number of different alignments.

Reply:
https://github.com/nasa/cFE/issues/1452

Good point, it makes the structure divisible by 64 such that the compiler won't add padding if the payload structure requires 64 bit alignment.